src/Client.ts

```diff
@@ -446,64 +434,47 @@ export default class Client {
     if (!opts) {
       opts = {}
     }
-    if (!opts.startTime) {
-      opts.startTime = new Date(0)
-    }
-    if (!opts.endTime) {
-      opts.endTime = new Date(new Date().toUTCString())
-    }
-    if (!opts.pageSize) {
-      opts.pageSize = 10
-    }
+    const { startTime, endTime, checkAddresses, limit } = opts
 
-    let wakuMsgs = await this.waku.store.queryHistory([topic], {
-      pageSize: opts.pageSize,
-      pageDirection: PageDirection.FORWARD,
-      timeFilter: {
-        startTime: opts.startTime,
-        endTime: opts.endTime,
-      },
-    })
-    wakuMsgs = wakuMsgs.filter((wakuMsg) => wakuMsg?.payload)
-    let msgs = await Promise.all(
-      wakuMsgs.map((wakuMsg) =>
-        this.decodeMessage(wakuMsg.payload as Uint8Array, wakuMsg.contentTopic)
-      )
+    const res = await this.apiClient.query(
+      { contentTopics: [topic], startTime, endTime },
+      {
+        direction: messageApi.SortDirection.SORT_DIRECTION_ASCENDING,
+        limit,
+      }
     )
-    if (opts?.checkAddresses) {
+
+    let msgs: Message[] = []
+    for (const env of res) {
+      if (!env.message) continue
+      try {
+        const msg = await this.decodeMessage(
+          b64Decode(env.message as unknown as string),
```

Comment:
Have to coerce the types here since the generated types for `UInt8Array` are a lie on the return value